lib/galaxy_test/api/test_history_contents.py

```diff
@@ -1026,6 +1026,73 @@ def test_purging_collection_should_purge_contents(self):
+        with self.dataset_populator.test_history() as history_id:
+            num_expected_collections = 1
+            num_expected_datasets = 2
+            collection_ids = self._create_collection_in_history(history_id, num_expected_collections)
+            original_collection_id = collection_ids[0]
+            # Check datasets are hidden and not deleted
+            history_contents = self._get_history_contents(history_id)
+            datasets = list(filter(lambda item: item["history_content_type"] == "dataset", history_contents))
+            assert len(datasets) == num_expected_datasets
+            for dataset in datasets:
+                assert dataset["deleted"] is False
+                assert dataset["visible"] is False
+
+            # Create a new list collection with the same datasets
+            payload = {
+                "name": "New list with same datasets",
+                "type": "dataset_collection",
+                "collection_type": "list",
+                "copy_elements": False,  # Reference the same datasets
+                "hide_source_items": True,
+                "element_identifiers": [
+                    {"name": "el1", "src": "hda", "id": datasets[0]["id"]},
+                    {"name": "el2", "src": "hda", "id": datasets[1]["id"]},
+                ],
+            }
+            create_response = self._post(f"histories/{history_id}/contents", payload, json=True)
+            self._assert_status_code_is_ok(create_response)
+            new_collection_id = create_response.json()["id"]
+
+            # Delete the original collection
+            payload = {
+                "operation": "delete",
+                "items": [
+                    {
+                        "id": original_collection_id,
+                        "history_content_type": "dataset_collection",
+                    },
+                ],
+            }
+            bulk_operation_result = self._apply_bulk_operation(history_id, payload)
+            self._assert_bulk_success(bulk_operation_result, 1)
+
+            # Check datasets are still not deleted
+            history_contents = self._get_history_contents(history_id)
+            for item in history_contents:
+                if item["history_content_type"] == "dataset":
+                    assert item["deleted"] is False
+
+            # Delete the new collection too
+            payload = {
+                "operation": "delete",
+                "items": [
+                    {
+                        "id": new_collection_id,
+                        "history_content_type": "dataset_collection",
+                    },
+                ],
+            }
+            bulk_operation_result = self._apply_bulk_operation(history_id, payload)
+            self._assert_bulk_success(bulk_operation_result, 1)
+
+            # Check datasets are deleted (and collections too)
+            history_contents = self._get_history_contents(history_id)
```

Comment:
same as above? So we can be reasonably certain that we are examining the same set of items at each step.